homeassistant/components/aurora/__init__.py

```diff
@@ -1 +1,142 @@
 """The aurora component."""
+
+import asyncio
+from datetime import timedelta
+import logging
+
+from auroranoaa import AuroraForecast
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_LATITUDE,
+    CONF_LONGITUDE,
+    CONF_NAME,
+    CONF_SCAN_INTERVAL,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import (
+    AURORA_API,
+    CONF_THRESHOLD,
+    COORDINATOR,
+    DEFAULT_POLLING_INTERVAL,
+    DEFAULT_THRESHOLD,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["binary_sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Aurora component."""
+    hass.data.setdefault(DOMAIN, {})
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Aurora from a config entry."""
+
+    conf = entry.data
+    options = entry.options
+
+    session = aiohttp_client.async_get_clientsession(hass)
+    api = AuroraForecast(session)
+
+    longitude = conf[CONF_LONGITUDE]
+    latitude = conf[CONF_LATITUDE]
+    polling_interval = options.get(CONF_SCAN_INTERVAL, DEFAULT_POLLING_INTERVAL)
+    threshold = options.get(CONF_THRESHOLD, DEFAULT_THRESHOLD)
+    name = conf[CONF_NAME]
+
+    try:
+        await api.get_forecast_data(longitude, latitude)
+    except ConnectionError:
+        return False
+    except Exception as error:  # pylint: disable=broad-except
+        raise ConfigEntryNotReady(error) from error
+
+    coordinator = AuroraDataUpdateCoordinator(
+        hass=hass,
+        name=name,
+        polling_interval=polling_interval,
+        api=api,
+        latitude=latitude,
+        longitude=longitude,
+        threshold=threshold,
+    )
+
+    await coordinator.async_refresh()
```

Comment:
You're now fetching the data twice. Would you be able to use `coordinator.last_update_success` to check if something didn't go right?

Reply:
Oh, you already do that. Why do you also do the check above? That seems double, you should drop that one.

Reply:
I didn't realize that the coordinator fetches on init. I check for last_update_success immediately after, so I can remove this if that's the case. Can you confirm?

Reply:
I don't think I'm following exactly what you're asking. Do I need to remove the async_refresh or the last_update_success or both?